FYI:

home/buildnode/jenkins/workspace/swift-PR-Linux-smoke-test/branch-master/buildbot_linux/swift-linux-x86_64/bin/swift-autolink-extract
lit.py: /home/buildnode/jenkins/workspace/swift-PR-Linux-smoke-test/branch-master/llvm/utils/lit/lit/TestingConfig.py:101: fatal: unable to parse config file '/home/buildnode/jenkins/workspace/swift-PR-Linux-smoke-test/branch-master/swift/test/lit.cfg', traceback: Traceback (most recent call last):
  File "/home/buildnode/jenkins/workspace/swift-PR-Linux-smoke-test/branch-master/llvm/utils/lit/lit/TestingConfig.py", line 88, in load_from_path
    exec(compile(data, path, 'exec'), cfg_globals, None)
  File "/home/buildnode/jenkins/workspace/swift-PR-Linux-smoke-test/branch-master/swift/test/lit.cfg", line 941, in <module>
    runtime_libs['tsan'] = 'tsan_runtime'
TypeError: list indices must be integers, not strings

--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -925,20 +925,26 @@ source_compiler_rt_libs(os.path.join(test_resource_dir, 'clang', 'lib',
 
 def check_runtime_libs(features_to_check):
     for lib in config.compiler_rt_libs:
-        for (libname, feature) in features_to_check:
+        for (libname, feature) in features_to_check.iteritems():
             if lib.startswith("libclang_rt." + libname + "_"):
                 config.available_features.add(feature)
 
-runtime_libs = [
-    ('profile', 'profile_runtime'),
-    ('asan', 'asan_runtime'),
-    ('ubsan', 'ubsan_runtime'),
-    ('safestack', 'safestack_runtime'),
-    ('fuzzer', 'fuzzer_runtime')
-]
+runtime_libs = {
+    'profile': 'profile_runtime',
+    'asan': 'asan_runtime',
+    'ubsan': 'ubsan_runtime',
+    'safestack': 'safestack_runtime',
+    'fuzzer': 'fuzzer_runtime'
+}
 
 if run_ptrsize != "32":
-    runtime_libs.append(('tsan', 'tsan_runtime'))
+    runtime_libs['tsan'] = 'tsan_runtime'
+
+# TODO: remove these lines once sanitizers start working on Linux again:
+# Tracked in https://bugs.swift.org/browse/SR-6257.
+if config.target_sdk_name == "linux":
+    runtime_libs.pop('asan')
+    runtime_libs.pop('tsan')
 
 check_runtime_libs(runtime_libs)
 

